src/ceph-volume/ceph_volume/util/system.py

```diff
@@ -61,10 +63,35 @@ def _get_path(executable, locations):
         '/usr/sbin',
         '/sbin',
     )
-    exec_in_static_locations = _get_path(executable, static_locations)
-    if exec_in_static_locations:
-        mlogger.warning('Found executable under {}, please ensure $PATH is set correctly!'.format(exec_in_static_locations))
-        return exec_in_static_locations
+
+    if not run_on_host:
+        path = os.getenv('PATH', '')
+        path_locations = path.split(':')
+        exec_in_path = _get_path(executable, path_locations)
+        if exec_in_path:
+            return exec_in_path
+        mlogger.warning('Executable {} not in PATH: {}'.format(executable, path))
+
+        exec_in_static_locations = _get_path(executable, static_locations)
+        if exec_in_static_locations:
+            mlogger.warning('Found executable under {}, please ensure $PATH is set correctly!'.format(exec_in_static_locations))
+            return exec_in_static_locations
+    else:
+        for location in static_locations:
+            path = '{}/{}'.format(location, executable)
+            command = []
+            command.extend(run_host_cmd + ['/usr/bin/stat', path])
+
+            process = subprocess.Popen(
+                command,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                stdin=subprocess.PIPE,
+                close_fds=True
+            )
+            returncode = process.wait()
+            if returncode == 0:
+                return path
```

Comment:
Is it possible to just check them all at once rather than having to loop through each file individually. E.g. If I made local dirs called bin, sbin and usr/sbin and put a dummy "executable" in just sbin I can stat on all at once and it will tell me which actually exists and which doesn't.

```
bash-5.0$ ls
bin  sbin  usr
bash-5.0$ stat bin/actual_executable sbin/actual_executable usr/sbin/actual_executable
stat: cannot stat 'bin/actual_executable': No such file or directory
  File: sbin/actual_executable
  Size: 7         	Blocks: 8          IO Block: 4096   regular file
Device: fd03h/64771d	Inode: 532538      Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/  adking)   Gid: ( 1000/  adking)
Access: 2022-01-13 08:08:53.012232661 -0500
Modify: 2022-01-13 08:08:53.012232661 -0500
Change: 2022-01-13 08:08:53.012232661 -0500
 Birth: 2022-01-13 08:08:53.012232661 -0500
stat: cannot stat 'usr/sbin/actual_executable': No such file or directory
```
Could save us having to run an nsenter command 6 times just to find the file with a bit of output parsing.

Reply:
If you do decide to do this I would say just create a standalone "check_files_exist_on_host" function that can take a list of files and return which ones exist on the host in case this comes up again.

Reply:
There's also the issue that the location of stat could possibly be chicken-and-egg-y what if distro X starts putting stat at /bin/  for example. This is probably fine for now, but I think longer term @adk3798 suggestion is on the right track and there should be a small lib of calls that are clearly for working on the host from w/in a (potentially privileged) container.

Reply:
Yeh, I had this concern too.
@adk3798's suggestion is a great idea, I'm going to address it in this PR.